docs/content/docs/concepts/keypairs.mdx

```diff
@@ -206,29 +206,174 @@ Private keys can be used by their owner to produce a digital signature of a spec
 
-### Verifying messages
+### Verifying signatures
 
 The public key associated with a given private key can be used by anyone to verify that a message was signed by the holder of the associated private key, as described above. The verification function requires the original message, a digital signature, and the public key associated with the owner who is believed to have produced that signature. Successful verification implies that the contents of the message signed by the owner are identical to the contents of the message that was given to the verification function. Key owners are free to share their public key with anyone they would like to grant the ability to verify their digital signatures.
 
-TODO
+If someone sends us a message and signature that they claim to be from the example above, we could use Bob's public key to verify that the signature was in fact the one produced by Bob and that the message was not modified in transit.
+
+```ts twoslash
+import { SignatureBytes } from "@solana/kit";
+const bobsPublicKey = null as unknown as CryptoKey;
+const supposedlyBobsSignature = null as unknown as SignatureBytes;
+// ---cut-before---
+import { verifySignature } from "@solana/kit";
+const verificationSucceeded = await verifySignature(
+  bobsPublicKey,
+  supposedlyBobsSignature,
+  new TextEncoder().encode("The meeting is at 6:00pm"),
+);
+if (!verificationSucceeded) {
+  throw new Error("Either the message was modified, the signature was not produced by Bob, or both");
+}
+```
+
+Verification protects equally against false claims of who produced the signature as it does against false claims about what message was signed. Both of these will return `false`;
+
+```ts twoslash
+import { SignatureBytes, verifySignature } from "@solana/kit";
+const bobsPublicKey = null as unknown as CryptoKey;
+const bobsSignature = null as unknown as SignatureBytes;
+const mallorysSignature = null as unknown as SignatureBytes;
+const signature = null as unknown as SignatureBytes;
+const textEncoder = new TextEncoder();
+const encode = textEncoder.encode.bind(textEncoder);
+// ---cut-before---
+// False claim that Bob signed the message when it was in fact Mallory
+await verifySignature(bobsPublicKey, mallorysSignature, encode("The meeting is at 6:00pm"));
+//                                   ^^^^^^^^^^^^^^^^^
+// False claim about the contents of the message
+await verifySignature(bobsPublicKey, bobsSignature, encode("The meeting is at 6:00am"));
+//                                                                            ^^^^^^
+```
 
 ### Deriving addresses
 
-TODO
+The Solana address associated with any public key can be derived using the [`getAddressFromPublicKey()`](/api/functions/getAddressFromPublicKey) function.
+
+```ts twoslash
+const publicKey = null as unknown as CryptoKey;
+// ---cut-before---
+import { getAddressFromPublicKey } from "@solana/kit";
+const address = await getAddressFromPublicKey(publicKey);
+```
+
+## Signatures
```

Comment:
Let me know if you think this entire section is necessary. I didn't know whether this was too much detail or if it's actually important to go over the guard/assertion functions for every type in the library.